File: RokidSDKDemoJava/app/src/main/java/com/rokid/mobile/sdk/demo/java/base/item/CardItem.java
package com.rokid.mobile.sdk.demo.java.base.item;

import android.view.View;
import android.widget.TextView;

import com.rokid.mobile.lib.entity.bean.channel.CardMsgBean;
import com.rokid.mobile.sdk.demo.java.R;
import com.rokid.mobile.ui.recyclerview.item.BaseItem;
import com.rokid.mobile.ui.recyclerview.item.BaseViewHolder;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by tt on 2018/2/24.
 */

public class CardItem extends BaseItem<CardMsgBean> {

    @BindView(R.id.fragment_msg_id)
    TextView deviceId;

    @BindView(R.id.fragment_card_content)
    TextView contentTxt;

    public CardItem(CardMsgBean data) {
        super(data);
    }

    @Override
    public int getViewType() {
        return 0;
    }

    @Override
    public int getLayoutId(int viewType) {
        return R.layout.item_card;
    }

    @Override
    public void onReleaseViews(BaseViewHolder holder, int sectionKey, int sectionViewPosition) {
        ButterKnife.bind(this, holder.getItemView());
        deviceId.setText("");
        deviceId.setVisibility(View.GONE);
        contentTxt.setVisibility(View.GONE);
    }

    @Override
    public void onSetViewsData(BaseViewHolder holder, int sectionKey, int sectionViewPosition) {
        deviceId.setText(getData().getMsgId());
        contentTxt.setText(getData().getMsgTxt());
        deviceId.setVisibility(View.VISIBLE);
        contentTxt.setVisibility(View.VISIBLE);
    }

}

File: RokidSDKDemokotlin/app/src/main/kotlin/com/rokid/mobile/sdk/demo/base/item/CardItem.kt
package com.rokid.mobile.sdk.demo.base.item

import com.rokid.mobile.lib.entity.bean.remotechannel.CardMsgBean
import com.rokid.mobile.sdk.demo.R
import com.rokid.mobile.ui.recyclerview.item.BaseItem
import com.rokid.mobile.ui.recyclerview.item.BaseViewHolder
import kotlinx.android.synthetic.main.item_card.view.*

/**
 * @author hongquan.zhao
 * Create Date:2018/3/21
 * Version:V1.0
 */

class CardItem(data: CardMsgBean) : BaseItem<CardMsgBean>(data) {

    val DeviceItem_ITEM_TYPE = 0

    override fun getViewType(): Int = DeviceItem_ITEM_TYPE

    override fun getLayoutId(viewType: Int): Int = R.layout.item_card

    override fun onReleaseViews(holder: BaseViewHolder?, sectionKey: Int, sectionViewPosition: Int) {
        holder!!.itemView.apply {
            fragment_msg_id.text = ""
            fragment_card_content.text = ""
        }
    }

    override fun onSetViewsData(holder: BaseViewHolder?, sectionKey: Int, sectionViewPosition: Int) {
        holder!!.itemView.apply {
            fragment_msg_id.text = data.msgId
            fragment_card_content.text = data.msgTxt
        }
    }

}